queue/src/main/java/org/killbill/queue/DBBackedQueueWithPolling.java

```diff
@@ -0,0 +1,220 @@
+        if (config.getPersistentQueueMode() == PersistentQueueMode.STICKY_POLLING) {
+            for (final T entry : entriesLeftBehind) {
+                entry.setCreatedDate(now);
+                entry.setProcessingState(PersistentQueueEntryLifecycleState.AVAILABLE);
+                entry.setCreatingOwner(CreatorName.get());
+            }
+            transactional.insertEntries(entriesLeftBehind, config.getTableName());
+        }
+    }
+
+    private List<T> fetchReadyEntries(int maxEntries) {
+        final Date now = clock.getUTCNow().toDate();
+        final String owner = config.getPersistentQueueMode() == PersistentQueueMode.POLLING ? null : CreatorName.get();
+        return executeQuery(new Query<List<T>, QueueSqlDao<T>>() {
+            @Override
+            public List<T> execute(final QueueSqlDao<T> queueSqlDao) {
+
+                long ini = System.nanoTime();
+                final List<T> result = queueSqlDao.getReadyEntries(now, maxEntries, owner, config.getTableName());
+                rawGetEntriesTime.update(System.nanoTime() - ini, TimeUnit.NANOSECONDS);
+                return result;
+            }
+        });
+    }
+
+
+    private List<T> claimEntries(final List<T> candidates) {
+        switch (config.getPersistentQueueMode()) {
+            case POLLING:
+                return sequentialClaimEntries(candidates);
+
+            case STICKY_POLLING:
+                return batchClaimEntries(candidates);
+
+            default:
+                throw new IllegalStateException("Unsupported PersistentQueueMode " + config.getPersistentQueueMode());
+        }
+    }
+
+    private List<T> batchClaimEntries(final List<T> candidates) {
+        if (candidates.isEmpty()) {
+            return ImmutableList.of();
+        }
+        final Date nextAvailable = clock.getUTCNow().plus(config.getClaimedTime().getMillis()).toDate();
+        final Collection<Long> recordIds = Collections2.transform(candidates, new Function<T, Long>() {
+            @Override
+            public Long apply(final T input) {
+                return input.getRecordId();
+            }
+        });
+        final int resultCount = executeQuery(new Query<Integer, QueueSqlDao<T>>() {
+            @Override
+            public Integer execute(final QueueSqlDao<T> queueSqlDao) {
+                long ini = System.nanoTime();
+                final Integer result = queueSqlDao.claimEntries(recordIds, clock.getUTCNow().toDate(), CreatorName.get(), nextAvailable, config.getTableName());
+                rawClaimEntriesTime.update(System.nanoTime() - ini, TimeUnit.NANOSECONDS);
+                return result;
+            }
+        });
+        // We should ALWAYS see the same number since we are in STICKY_POLLING mode and there is only one thread claiming entries.
+        // We keep the 2 cases below for safety (code was written when this was MT-threaded), and we log with warn (will eventually remove it in the future)
```

Comment:
We should maybe start cleaning all of that stuff, it's a lot of codepaths 😅

Reply:
Ok, let's have the discussion - not 100% convinced, but i also like the idea of more simplicity.